From 9c1d0ce52d7939f43b91343e7e7eef5b35ce75fc Mon Sep 17 00:00:00 2001
From: Uri Mashiach <uri.mashiach@compulab.co.il>
Date: Sun, 24 Jan 2021 14:42:07 +0200
Subject: [PATCH 107/113] iot-gate-imx8: add support for the IE-CL420 module

The IE-CL420 implements one channel ADC.

The CAN, TPM and the CL420 modules are 3 alternatives in the board
IED-BASE, slots A and B for the SPI bus.
If CAN, TPM or CL420 module is detected:
* The relevant SPI bus node is enabled in the device tree.
* The relevant TPM, CAN, or ADC node is enabled in the device tree.

Signed-off-by: Uri Mashiach <uri.mashiach@compulab.co.il>
---
 board/compulab/iot-gate-imx8/iot-gate-imx8.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/board/compulab/iot-gate-imx8/iot-gate-imx8.c b/board/compulab/iot-gate-imx8/iot-gate-imx8.c
index 46c1838055..308dd91438 100644
--- a/board/compulab/iot-gate-imx8/iot-gate-imx8.c
+++ b/board/compulab/iot-gate-imx8/iot-gate-imx8.c
@@ -177,6 +177,7 @@ int sub_board_late_init(void)
 #define IOT_GATE_IMX8_CARD_ID_RS_485 2 /* Card ID - IED-RS485   */
 #define IOT_GATE_IMX8_CARD_ID_TPM    3 /* Card ID - IED-TPM     */
 #define IOT_GATE_IMX8_CARD_ID_CAN    4 /* Card ID - IED-CAN     */
+#define IOT_GATE_IMX8_CARD_ID_CL420  5 /* Card ID - IED-CL420   */
 #define IOT_GATE_IMX8_CARD_ID_RS_232 6 /* Card ID - IED-RS232   */
 #define IOT_GATE_IMX8_DTB_PATH_UART2 "/soc@0/bus@30800000/serial@30890000"
 #define IOT_GATE_IMX8_DTB_PATH_UART4 "/soc@0/bus@30800000/serial@30a60000"
@@ -186,6 +187,8 @@ int sub_board_late_init(void)
 #define IOT_GATE_IMX8_DTB_PATH_CAN1 "/soc@0/bus@30800000/spi@30840000/can@0"
 #define IOT_GATE_IMX8_DTB_PATH_TPM0 "/soc@0/bus@30800000/spi@30830000/tpm@0"
 #define IOT_GATE_IMX8_DTB_PATH_TPM1 "/soc@0/bus@30800000/spi@30840000/tpm@0"
+#define IOT_GATE_IMX8_DTB_PATH_ADC0 "/soc@0/bus@30800000/spi@30830000/adc@0"
+#define IOT_GATE_IMX8_DTB_PATH_ADC1 "/soc@0/bus@30800000/spi@30840000/adc@0"
 
 /*
  * iot_gate_imx8_update_ext_ied()
@@ -242,6 +245,7 @@ static int iot_gate_imx8_update_ext_ied(void *blob)
 		break;
 	case IOT_GATE_IMX8_CARD_ID_CAN:
 	case IOT_GATE_IMX8_CARD_ID_TPM:
+	case IOT_GATE_IMX8_CARD_ID_CL420:
 		do_fixup_by_path(blob, IOT_GATE_IMX8_DTB_PATH_ECSPI2, "status",
 				 "okay", sizeof("okay"), 1);
 		if (revision == IOT_GATE_IMX8_CARD_ID_CAN)
@@ -250,6 +254,9 @@ static int iot_gate_imx8_update_ext_ied(void *blob)
 		else if (revision == IOT_GATE_IMX8_CARD_ID_TPM)
 			do_fixup_by_path(blob, IOT_GATE_IMX8_DTB_PATH_TPM0, "status",
 					 "okay", sizeof("okay"), 1);
+		else if (revision == IOT_GATE_IMX8_CARD_ID_CL420)
+			do_fixup_by_path(blob, IOT_GATE_IMX8_DTB_PATH_ADC0, "status",
+					 "okay", sizeof("okay"), 1);
 		break;
 	default:
 		printf("%s: invalid slot 0 card ID: %d\n", __func__, revision);
@@ -272,6 +279,7 @@ static int iot_gate_imx8_update_ext_ied(void *blob)
 		break;
 	case IOT_GATE_IMX8_CARD_ID_CAN:
 	case IOT_GATE_IMX8_CARD_ID_TPM:
+	case IOT_GATE_IMX8_CARD_ID_CL420:
 		do_fixup_by_path(blob, IOT_GATE_IMX8_DTB_PATH_ECSPI3, "status",
 				 "okay", sizeof("okay"), 1);
 		if (revision == IOT_GATE_IMX8_CARD_ID_CAN)
@@ -280,6 +288,9 @@ static int iot_gate_imx8_update_ext_ied(void *blob)
 		else if (revision == IOT_GATE_IMX8_CARD_ID_TPM)
 			do_fixup_by_path(blob, IOT_GATE_IMX8_DTB_PATH_TPM1, "status",
 					 "okay", sizeof("okay"), 1);
+		else if (revision == IOT_GATE_IMX8_CARD_ID_CL420)
+			do_fixup_by_path(blob, IOT_GATE_IMX8_DTB_PATH_ADC1, "status",
+					 "okay", sizeof("okay"), 1);
 		break;
 	default:
 		printf("%s: invalid slot 1 card ID: %d\n", __func__, revision);
-- 
2.11.0

